Unable to compile in OpenCV 4.2

Hi @sthoduka,

Thank you for providing and maintaining this reference code. It has helped me understand better how Fourier Mellin transform works and how robust it can be given noise in image evaluation. :smile: 

## Issue Description
> I do note that the code is meant to compile only with OpenCV 2.4. Therefore, this issue is only created to justify a pull request I will be making shortly. 

When attempting to compile the code using OpenCV 4.2, it generates the following error.
```bash
# Compiling the code
mkdir build && cd build
cmake -DCMAKE_BUILD_TYPE=Release ..
make
```

```bash
/home/user/imreg_fmt/src/image_transforms.cpp: In member function ‘void ImageTransforms::remapLogPolar(const cv::Mat&, cv::Mat&)’:
/home/user/imreg_fmt/src/image_transforms.cpp:52:43: error: ‘CV_INTER_CUBIC’ was not declared in this scope
   52 |     cv::remap(src, dst, cv_xMap, cv_yMap, CV_INTER_CUBIC & cv::INTER_MAX, cv::BORDER_CONSTANT, cv::Scalar());

```

## Cause of Issue
This build failure is caused by the fact I am using OpenCV 4.2 and therefore [some MACROS in OpenCV 2.4 are not present in newer versions](https://answers.ros.org/question/214031/catkin_make-error-cv_inter_linear-was-not-declared-in-this-scope/?answer=371308#post-id-371308).

This issue is addressed by Pull Request #7.

--- a/README.md
+++ b/README.md
@@ -9,13 +9,12 @@ For images of size 320x240, the algorithm runs at approximately 14 Hz on an Inte
 Dependencies
 ------------
 * fftw3
-* OpenCV 2.4
+* OpenCV 4.2
 * Eigen
 
 Compile
 -------
-    mkdir build
-    cd build
+    mkdir build && cd build
     cmake -DCMAKE_BUILD_TYPE=Release ..
     make
 

--- a/src/image_main.cpp
+++ b/src/image_main.cpp
@@ -12,8 +12,8 @@ int main(int argc, char **argv)
         std::cout << "specify two image files" << std::endl;
         return 1;
     }
-    im0 = cv::imread(argv[1], CV_LOAD_IMAGE_COLOR);
-    im1 = cv::imread(argv[2], CV_LOAD_IMAGE_COLOR);
+    im0 = cv::imread(argv[1], cv::IMREAD_COLOR);
+    im1 = cv::imread(argv[2], cv::IMREAD_COLOR);
 
     cv::imshow("im0", im0);
     cv::imshow("im1", im1);

--- a/src/image_registration.cpp
+++ b/src/image_registration.cpp
@@ -66,7 +66,7 @@ cv::Mat ImageRegistration::getBorderMask()
 void ImageRegistration::processImage(const cv::Mat &im, cv::Mat &gray, cv::Mat &log_polar)
 {
     im.convertTo(gray, CV_32F, 1.0/255.0);
-    cv::cvtColor(gray, gray, CV_BGR2GRAY);
+    cv::cvtColor(gray, gray, cv::COLOR_BGR2GRAY);
 
     cv::Mat apodized;
     cv::Mat im_dft_cv;

--- a/src/image_transforms.cpp
+++ b/src/image_transforms.cpp
@@ -49,7 +49,7 @@ void ImageTransforms::createLogPolarMap()
 
 void ImageTransforms::remapLogPolar(const cv::Mat &src, cv::Mat &dst)
 {
-    cv::remap(src, dst, cv_xMap, cv_yMap, CV_INTER_CUBIC & cv::INTER_MAX, cv::BORDER_CONSTANT, cv::Scalar());
+    cv::remap(src, dst, cv_xMap, cv_yMap, cv::INTER_CUBIC & cv::INTER_MAX, cv::BORDER_CONSTANT, cv::Scalar());
 }
 
 void ImageTransforms::getScaleRotation(double row, double col, double &scale, double &rotation)
@@ -114,4 +114,3 @@ Eigen::MatrixXd ImageTransforms::getApodizationWindow(int rows, int cols, int ra
 
     return col * row.transpose();
 }
-

